### DISCUSSION TITLE
Will new extension eventually include testing functionality?

### DISCUSSION THREAD

**Discussion by lonix1** [OmniSharp/omnisharp-vscode#6457] [General]
The new extension's development is really encouraging, and the gap in features between the old and new extensions is narrowing rapidly.

Testing has been a sore point for c# in vscode since the beginning. There is the community-supported [.NET Core Test Explorer](https://github.com/formulahendry/vscode-dotnet-test-explorer), but it's clunky, riddled with problems and hard to maintain.

Do you have plans to also include testing capabilities?

**Comment by dibarbet** [comment]
C# Dev Kit has full integration with VSCode's test explorer, see https://code.visualstudio.com/docs/csharp/testing

The C# extension standalone has some limited testing functionality via code lens.  See also https://github.com/dotnet/vscode-csharp/issues/5719.  We don't currently have plans to bring test explorer integration to C# standalone.

**Reply by kl-tie** [reply]
@dibarbet Any reasons why limiting such important feature like test explorer in C#? Most programming languages & frameworks provides test explorer integration without non-FOSS compliant license such as Ruby LSP and Dart.

It seems toolings in C# is very limited and too much constraint due to C# Dev Kit license, this is not what you would call .NET is FOSS friendly especially when the basic and important toolings such test explorer itself is not provided for free.
